lib/apiservers/engine/backends/convert/stats.go

```diff
@@ -205,10 +213,8 @@ func (cs *ContainerStats) Listen() *io.PipeWriter {
 func (cs *ContainerStats) ToContainerStats(current *metrics.VMMetrics) (*types.StatsJSON, error) {
 	// if we have a current metric then validate and transform
 	if cs.currentMetric != nil {
-		// do we have the same metric as before?
-		if cs.currentMetric.SampleTime.Equal(current.SampleTime) {
-			// we've already got this as current, so skip and wait for the
-			// next sample
+		// do we have the same metric or has the metric not been initialized?
+		if cs.currentMetric.SampleTime.Equal(current.SampleTime) || current.SampleTime.IsZero() {
```

Comment:
yes - at the end of a stats call there was an intermittent issue where a final metric would be produced, but it wasn't completely initialized.   The easiest way to tell is the fact that the sampleTime wasn't set.